drools-wb-screens/drools-wb-test-scenario-editor/drools-wb-test-scenario-editor-client/src/main/java/org/drools/workbench/screens/testscenario/client/reporting/TestRunnerReportingViewImpl.java

```diff
@@ -20,120 +20,130 @@
+                                   }
+                               });
     }
 
-    private String makeMessage( Failure failure ) {
+    private String makeMessage(Failure failure) {
         final String displayName = failure.getDisplayName();
         final String message = failure.getMessage();
-        return displayName + ( !( message == null || message.isEmpty() ) ? " : " + message : "" );
+        return displayName + (!(message == null || message.isEmpty()) ? " : " + message : "");
     }
 
     @Override
-    public void setPresenter( Presenter presenter ) {
+    public void setPresenter(Presenter presenter) {
         this.presenter = presenter;
     }
 
     @Override
-    public void bindDataGridToService( TestRuntimeReportingService testRuntimeReportingService ) {
-        testRuntimeReportingService.addDataDisplay( dataGrid );
+    public void bindDataGridToService(TestRuntimeReportingService testRuntimeReportingService) {
+        testRuntimeReportingService.addDataDisplay(dataGrid);
     }
 
     @Override
     public void showSuccess() {
-        successPanel.setVisible( true );
-        failurePanel.setVisible( false );
+        resultTitle.textContent = TestScenarioConstants.INSTANCE.Success();
+        resultTitle.className = "label col-md-12 label-success";
     }
 
     @Override
     public void showFailure() {
-        failurePanel.setVisible( true );
-        successPanel.setVisible( false );
+        resultTitle.textContent = TestScenarioConstants.INSTANCE.ThereWereTestFailures();
+        resultTitle.className = "label col-md-12 label-danger";
     }
 
     @Override
-    public void setExplanation( String explanation ) {
+    public void setExplanation(String explanation) {
     }
 
     @Override
-    public void setRunStatus( int runCount,
-                              long runTime ) {
-        Date date = new Date( runTime );
-        DateTimeFormat minutesFormat = DateTimeFormat.getFormat( "m" );
-        DateTimeFormat secondsFormat = DateTimeFormat.getFormat( "s" );
+    public void setRunStatus(int runCount,
+                             long runTime) {
+        Date date = new Date(runTime);
+        DateTimeFormat minutesFormat = DateTimeFormat.getFormat("m");
+        DateTimeFormat secondsFormat = DateTimeFormat.getFormat("s");
+
+        resultStats.textContent = TestScenarioConstants.INSTANCE.XTestsRanInYMinutesZSeconds(runCount,
+                                                                                             minutesFormat.format(date),
+                                                                                             secondsFormat.format(date));
+    }
 
-        stats.setText( TestScenarioConstants.INSTANCE.XTestsRanInYMinutesZSeconds( runCount, minutesFormat.format( date ), secondsFormat.format( date ) ) );
+    @Override
+    public Widget asWidget() {
+        return ElementWrapperWidget.getWidget(resultPanel);
     }
```

Comment:
I think our `TestRunnerReportingView` interface should extend `org.uberfire.client.mvp.UberElemental` and then we can omit this `asWidget()`

See [this class](https://github.com/kiegroup/kie-wb-common/blob/master/kie-wb-common-widgets/kie-wb-common-ui/src/main/java/org/kie/workbench/common/widgets/client/popups/validation/ValidationPopupView.java) for example.

Reply:
This is true. The reason for this is that the Presenter should not have GWT Widgets or components in it and we hide them behind interfaces like IsWidget or in the new parts of the code we should use UberElemental. The presenter should be testable without the GWT mockito runner.

Reply:
@jomarko on the other hand, you have to override getElement and init methods. But if it is the "new" way, I can do this.